index.js

```diff
@@ -1,28 +1,74 @@
-
-var s3ToLogstore = require('s3-to-logstore');
-var winston = require('winston');
-require('winston-elasticsearch');
+var aws  = require('aws-sdk');
+var zlib = require('zlib');
+var async = require('async');
+const CloudFrontParser = require('cloudfront-log-parser');
 
 var elasticsearch = require('elasticsearch');
 
+var s3 = new aws.S3();
+
 var client = new elasticsearch.Client({
   host: process.env.ES_HOST,
-  log: 'trace'
+  log: 'trace',
+  keepAlive: false
 });
 
-var transport = new winston.transports.Elasticsearch({
-  indexPrefix: process.env.ES_INDEXPREFIX,
-  client: client
-});
+exports.handler = function(event, context, callback) {
+    var srcBucket = event.Records[0].s3.bucket.name;
```

Comment:
IMHO, I believe it is impossible for Records to be empty. Since it belongs to `event`, which is the notification object received from S3. `event` being empty would mean that the lambda has not been called. 
 https://docs.aws.amazon.com/lambda/latest/dg/eventsources.html#eventsources-s3-put